### PR TITLE
#1332: Implement auto use of cycle-point format in a remote suite.

### LINKED ISSUE #1332: cylc suite-state command and cycle point format.
cylc-6 has somewhat broken automatic suite state polling tasks, which are defined like this in the graph:

```
polling_task<REMOTE_SUITE::REMOTE_TASK> => get-data
```

because the automatically-generated `cylc suite-state` command scripting assumes that the remote suite has the same cycle point format as the local one - which, from NIWA experience, is often not the case.

So, it would be good if `cylc suite-state` could take cycle point format as an optional argument, or for automatic use as above, from the environment.

### PR DESCRIPTION
**Pull request by arjclark**
Closes #1332

### CODE REVIEW

File: lib/cylc/dbstatecheck.py [hunk 1]
```diff
@@ -76,6 +76,15 @@ def display_maps(self, res):
             for row in res:
                 sys.stdout.write((", ").join(row).encode("utf-8") + "\n")
 
+    def get_remote_point_format(self):
```
Comment (matthewrmshin) [inline lib/cylc/dbstatecheck.py]: Should add doc string.

File: tests/suite-state/06-format.t [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#-------------------------------------------------------------------------------
+set_test_number 2
+#-------------------------------------------------------------------------------
+install_suite $TEST_NAME_BASE format
+#-------------------------------------------------------------------------------
+TEST_NAME=$TEST_NAME_BASE-run
+suite_run_ok $TEST_NAME cylc run --debug $SUITE_NAME
+#-------------------------------------------------------------------------------
+TEST_NAME=$TEST_NAME_BASE-cli-template
+run_ok $TEST_NAME cylc suite-state $SUITE_NAME -p 20100101T0000Z \
```
Comment (matthewrmshin) [inline tests/suite-state/06-format.t]: Maybe worth adding a test for STDOUT?

File: lib/cylc/dbstatecheck.py (hunk 1 above)
Comment (arjclark) [inline lib/cylc/dbstatecheck.py]: done

**Comment by arjclark** [comment]
@matthewrmshin - comments addressed

File: lib/cylc/task_pool.py [hunk 3]
```diff
@@ -1430,6 +1430,11 @@ def put_rundb_suite_params(self, initial_point, final_point):
             {"key": "initial_point", "value": str(initial_point)},
             {"key": "final_point", "value": str(final_point)},
         ])
+        if format:
+            print "during format thing"
```
Comment (hjoliver) [inline lib/cylc/task_pool.py]: A stray bit of debug code? :grin:

Comment (arjclark) [inline lib/cylc/task_pool.py]: Well spotted, fixed now.

**Comment by arjclark** [comment]
@hjoliver - spurious print removed :grin:

**Pull request merged**